lib/dashlib.py

```diff
@@ -176,6 +193,9 @@ def create_superblock(proposals, event_block_height, budget_max, sb_epoch_time):
     )
     printdbg("generated superblock: %s" % sb.__dict__)
 
+    if testnet:
+        assert len(sb.dashd_serialise()) == data_length
```

Comment:
I'm not sure about whether this is needed... data_length is an internal (to this method) counter, and also based on assumptions about serialization size. This also adds an un-necessary parameter to `create_superblock`.

I'd rather use use the serialization and test for that each time than keep track using `data_length`, since that's an extra item that has to be tracked along with the serialized value, and it's possible they could skew (for example, if the serialization format changes).

Reply:
OK. Probably we don't need this optimization and it would be better to keep things simple.